rust/index/src/spann/types.rs

```diff
@@ -0,0 +1,214 @@
+use std::collections::HashMap;
+
+use chroma_blockstore::{provider::BlockfileProvider, BlockfileWriter, BlockfileWriterOptions};
+use chroma_distance::DistanceFunction;
+use chroma_error::{ChromaError, ErrorCodes};
+use chroma_types::{CollectionUuid, SpannPostingList};
+use thiserror::Error;
+use uuid::Uuid;
+
+use crate::{
+    hnsw_provider::{HnswIndexProvider, HnswIndexRef},
+    IndexUuid,
+};
+
+// TODO(Sanket): Add locking structures as necessary.
+#[allow(dead_code)]
+pub struct SpannIndexWriter {
+    // HNSW index and its provider for centroid search.
+    hnsw_index: HnswIndexRef,
+    hnsw_provider: HnswIndexProvider,
+    // Posting list of the centroids.
+    // The blockfile also contains next id for the head.
```

Comment:
How does it contain the next id? is it a special key?

Reply:
I removed it in the next PR. The next head id is a separate blockfile